tests/grids/test_mortar_grid.py

```diff
@@ -0,0 +1,1094 @@
+
+    assert np.abs(new_projection_avg[0, 8] - 0.4 < 1e-6)
+    assert np.abs(new_projection_avg[0, 9] - 0.6 < 1e-6)
+    # The second cell in mortar grid is still fully connected to face 9
+    assert np.abs(new_projection_avg[1, 9] - 1 < 1e-6)
+    assert np.abs(new_projection_avg[2, 13] - 1 < 1e-6)
+    assert np.abs(new_projection_avg[3, 12] - 0.4 < 1e-6)
+    assert np.abs(new_projection_avg[3, 13] - 0.6 < 1e-6)
+
+
+def _create_2d_mdg(
+    size: list[int, int]
+) -> tuple[pp.MixedDimensionalGrid, pp.Grid, sps.spmatrix]:
+    """Helper function to create a 2d md grid with a single interface.
+
+    Parameters:
+        size: Number of cells in the x and y direction.
+
+    Returns:
+        mdg: The mixed-dimensional grid.
+        sd: The subdomain grid.
+        projection: The projection from the subdomain to the mortar grid.
+
+    """
+    # Create the new grid
+    mdg, _ = pp.md_grids_2d.single_horizontal(size, simplex=False)
+    # Fetch the interface and a projection matrix. The grid is matching, so we
+    # arbitrarily use the integration projection.
+    intf = mdg.interfaces()[0]
+    projection = intf.primary_to_mortar_int().copy()
+    sd = mdg.subdomains(dim=2)[0]
+    return mdg, sd, projection
+
+
+def _replace_2d_grid_fetch_projections(mdg, sd_old, sd_new, old_projection):
+    """Helper function to replace a 2d grid in a md grid, and fetch the new projections.
+
+    The function also does a small sanity check on the size of the projections.
+
+    Parameters:
+        mdg: The mixed-dimensional grid.
+        sd_old: The subdomain grid to be replaced.
+        sd_new: The new subdomain grid.
+        old_projection: The projection from the old subdomain grid to the mortar grid.
+
+    Returns:
+        new_projection_avg: The new projection from the new subdomain grid to the mortar
+            grid, for quantities which should be averaged.
+        new_projection_int: The new projection from the new subdomain grid to the mortar
+            grid, for quantities which should be summed.
+
+    """
+    # Do the replacement
+    mdg.replace_subdomains_and_interfaces({sd_old: sd_new})
+    # Get mortar grid
+    intf_new = mdg.interfaces()[0]
+
+    # Fetch the new projections
+    new_projection_avg = intf_new.primary_to_mortar_avg()
+    new_projection_int = intf_new.primary_to_mortar_int()
+
+    # Sanity check: The mortar grid is not changed, so the number of rows in the
+    # projection should not change. The number of columns should be the number of faces
+    # in the new subdomain grid.
+    assert new_projection_avg.shape[0] == old_projection.shape[0]
+    assert new_projection_avg.shape[1] == sd_new.num_faces
+
+    return new_projection_avg, new_projection_int
+
+
+"""Various tests for replacing subdomain and interface grids in a 3d mixed-
+dimensional grid.
+
+This can be considered integration tests for the method
+replace_subdomains_and_interfaces in the MixedDimensionalGrid class.
```

Comment:
In that case, it should be placed in test_md_grid.py, shouldn't it?

Reply:
Sorry, the comment was misleading. I will update.